### ISSUE TITLE
Template bug and usage of IOptions vs IOptionsMonitor

### ISSUE THREAD

**Issue by Ponant** [AzureAD/microsoft-identity-web#1473]
The template `dotnet new webapp --auth IndividualB2C` on .Net 6 RC1 provides a `_LoginPartial` that never shows the Edit link.

The culprit is the use of `IOptions<MicrosoftidentityOptions>` which returns null entries, for example `EditProfilePolicyId` or reset  policies are null, hence never shown on the browser:

````cshtml
@inject IOptions<MicrosoftIdentityOptions> AzureADB2COptions

@{
    var options = AzureADB2COptions.Value;
}
<ul class="navbar-nav">
@if (User.Identity?.IsAuthenticated == true)
{
        @if (!string.IsNullOrEmpty(options.EditProfilePolicyId))
        {
            <li class="nav-item">
                <a class="nav-link text-dark" asp-area="MicrosoftIdentity"
                     asp-controller="Account" asp-action="EditProfile">
                    <span class="text-dark">Hello @User.Identity?.Name!</span>
                </a>
            </li>
        }
        else
        {
            <li class="nav-item">
                <span class="navbar-text text-dark">Hello @User.Identity?.Name!</span>
            </li>
        }
        <li class="nav-item">
            <a class="nav-link text-dark" asp-area="MicrosoftIdentity"
                 asp-controller="Account" asp-action="SignOut">Sign out</a>
        </li>
````

#### Solution

Use `IOptionsMonitor` instead, e.g.

````cshtml
@inject IOptionsMonitor<MicrosoftIdentityOptions> AzureADB2COptions
@{
      var options = AzureADB2COptions.Get("OpenIdConnect");
      var reset =options.ResetPasswordPolicyId;
      var edit = options.EditProfilePolicyId;
}
````

#### Remark for a possible improvement

For what concerns at least webapps (no api calls), it seems there are two calls that look redundant, and one of them uses an unconfigured value:

```csharp
MergedOptions
.UpdateMergedOptionsFromMicrosoftIdentityOptions(msIdOptions.Value, mergedOptions);

MergedOptions
.UpdateMergedOptionsFromMicrosoftIdentityOptions(msIdOptionsMonitor.Get(openIdConnectScheme), mergedOptions);
````

Here https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs#L288-L299

But from my understanding the library does not configure for `IOptions`, only `IOptionsMonitor`, here
https://github.com/AzureAD/microsoft-identity-web/blob/626bd2c3a3a29cea6318edb354e6daf31abd24d3/src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs#L241

This is why it is not working in LoginPartial.

Unless I do not get a point, I think the first line can be removed

````csharp
MergedOptions
.UpdateMergedOptionsFromMicrosoftIdentityOptions(msIdOptions.Value, mergedOptions);
````

And the configure method in the link above simplified.

The `msIdOptions` is used only here and returns null values because it is not configured.

**version : 1.17.0**

**Comment by jennyf19** [comment]
I think @jmprieur fixed this for RC 2. We will need to check.

**Comment by jmprieur** [comment]
Thanks @Ponant 
This is fixed by https://github.com/dotnet/aspnetcore/pull/35811

**Comment by Ponant** [comment]
OK sorry I did not see this PR. And what about the double call https://github.com/AzureAD/microsoft-identity-web/blob/master/src/Microsoft.Identity.Web/WebAppExtensions/MicrosoftIdentityWebAppAuthenticationBuilderExtensions.cs#L288-L299 ?

**Comment by jennyf19** [comment]
This should be fixed in 2.5.0. If not, please re-open.
